go/client/cas_test.go

```diff
@@ -390,34 +376,42 @@ func TestWriteBlobs(t *testing.T) {
 			for _, tc := range tests {
 				t.Run(tc.name, func(t *testing.T) {
 					fake.Clear()
+					present := make(map[digest.Digest]bool)
 					for _, blob := range tc.present {
 						fake.Put(blob)
+						present[digest.NewFromBlob(blob)] = true
 					}
-					input := make(map[digest.Digest][]byte)
+					var input []*chunker.Chunker
 					for _, blob := range tc.input {
-						input[digest.NewFromBlob(blob)] = blob
+						input = append(input, chunker.NewFromBlob(blob, 20))
 					}
 
-					err := c.WriteBlobs(ctx, input)
+					err := c.Upload(ctx, input...)
 					if err != nil {
-						t.Fatalf("c.WriteBlobs(ctx, inputs) gave error %s, expected nil", err)
+						t.Fatalf("c.Upload(ctx, input) gave error %v, expected nil", err)
 					}
 
-					for _, blob := range tc.present {
-						dg := digest.NewFromBlob(blob)
-						if fake.BlobWrites(dg) > 0 {
-							t.Errorf("blob %v with digest %s was uploaded even though it was already present in the CAS", blob, dg)
+					for _, ch := range input {
+						dg := ch.Digest()
+						blob, err := ch.FullData()
+						if err != nil {
+							t.Fatalf("ch.FullData() returned an error: %v", err)
+						}
+						if present[dg] {
+							if fake.BlobWrites(dg) > 0 {
+								t.Errorf("blob %v with digest %s was uploaded even though it was already present in the CAS", blob, dg)
+							}
+							continue
```

Comment:
How do you make sure the blobs where actually uploaded and not only that the chunkers pulled all the data?

Reply:
fake.BlobWrites returns how many times Write or Batch upload was called on the digest. I don't actually know that the chunkers pulled all the data, but I verify in the next lines that the data is correct, so the conclusion is that it was uploaded.

Reply:
sgtm